modules/xfem/src/geomsearch/XFEMElementPairLocator.C

```diff
@@ -41,5 +46,41 @@ XFEMElementPairLocator::reinit()
 
     ElementPairInfo new_elem_info(elem, q_points, weights, normal);
     _element_pair_info.insert(std::pair<const Elem*, ElementPairInfo>(elem, new_elem_info));
+
+    // elem2
+    _xfem->getXFEMIntersectionInfo(elem2, plane_id, normal, intersectionPoints, _use_displaced_mesh);
+
+    if (intersectionPoints.size() == 2)
+      _xfem->getXFEMqRuleOnLine(intersectionPoints, q_points, weights);
+    else if (intersectionPoints.size() > 2)
+      _xfem->getXFEMqRuleOnSurface(intersectionPoints, q_points, weights);
+
+    ElementPairInfo new_elem_info2(elem2, q_points, weights, normal);
+    _element_pair_info.insert(std::pair<const Elem*, ElementPairInfo>(elem2, new_elem_info2));
```

Comment:
I think I'd just add a second set of data for the second element in there. Maybe something like:  `_neighbor_elem`, `_neighbor_constraint_q_point`, `_neighbor_constraint_JxW`, `_neighbor_normal`.